src/libraries/System.Net.Http.WinHttpHandler/tests/FunctionalTests/WinHttpHandlerTest.cs

```diff
@@ -100,36 +100,26 @@ public async Task SendAsync_SlowServerAndCancel_ThrowsTaskCanceledException()
         [Fact]
         public async Task SendAsync_SlowServerRespondsAfterDefaultReceiveTimeout_ThrowsHttpRequestException()
         {
-            var handler = new WinHttpHandler();
-            using (var client = new HttpClient(handler))
+            TaskCompletionSource<bool> tcs = new TaskCompletionSource<bool>();
+            await LoopbackServer.CreateClientAndServerAsync(async uri =>
             {
-                var triggerResponseWrite = new TaskCompletionSource<bool>();
-                var triggerRequestWait = new TaskCompletionSource<bool>();
-
-                await LoopbackServer.CreateServerAsync(async (server, url) =>
+                using var client = new HttpClient(new WinHttpHandler());
+                HttpRequestException ex = await Assert.ThrowsAsync<HttpRequestException>(() => client.GetAsync(uri));
+                _output.WriteLine($"Exception: {ex}");
```

Comment:
Is this PR to observe the exception or could this refactor somehow mitigate the issue? If it's the former, unfortunately I don't think there is more we can see than what is already reported in the issue. :(

Note that #102452 *might* be related. Do we know if this happens against Framework or .NET 9 runs?

Reply:
This PR just simplifies the code and removes unnecessary stuff. Not intended to fix it. I'll investigate this more.